machiavelli-cards/src/main/java/pl/jellytech/machiavelli/cards/services/CardServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package pl.jellytech.machiavelli.cards.services;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.jellytech.machiavelli.cards.entities.Card;
+import pl.jellytech.machiavelli.cards.repositories.CardRepository;
+
+import java.util.List;
+
+
+@Service
+@Slf4j
+public class CardServiceImpl implements CardService {
+    private final CardRepository cardRepository;
+    @Autowired
+    public CardServiceImpl(CardRepository cardRepository){
+        this.cardRepository = cardRepository;
+    }
+    @Transactional
+    public Card createOrUpdate(Card card){
+        this.cardRepository.save(card);
+        return card;
```

Comment:
you should always return the saved card, not the one made for the request since it doesn't confirm the save actually worked